block/manager.go

```diff
@@ -699,6 +699,13 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 		return err
 	}
 
+	// Check if the node has shutdown prior to publishing to channels
+	select {
+	case <-ctx.Done():
+		return nil
```

Comment:
since `nil` indicates successful scenario, node shutting down before publishing the block should return error instead or `nil` is still fine?

Reply:
I think you can look at it either way. 

The only actions after this step are signaling background loops. So at this point things have been successful. Deciding whether to quit early because of a shutdown, vs blocking on signaling background loops that are also shutting down seems like a successful scenario to me.